Problem: libzmq is built without security, by default

Previously, libzmq was always built with libsodium if possible. This is no longer the case.

This breaks the build process for upstream applications (e.g. CZMQ) and builds a fully insecure stack by default, which is not good (and a breaking change).

Worse, when you specify --with-libsodium, configure reports:

```
checking for sodium... no
configure: error: Package requirements (libsodium) were not met:
No package 'libsodium' found
```

Solution: implement existing contract, and always build with libsodium if it is found on the path. If a faster build is wanted, this should be an additional option (e.g. --with-no-security).

--- a/configure.ac
+++ b/configure.ac
@@ -395,7 +395,7 @@ have_sodium_library="no"
 AC_ARG_WITH([libsodium], [AS_HELP_STRING([--with-libsodium],
     [require libzmq build with libsodium. Requires pkg-config [default=no]])],
     [require_libsodium_ext=$withval],
-    [require_libsodium_ext=no])
+    [require_libsodium_ext=yes])
 
 # conditionally require libsodium package
 if test "x$require_libsodium_ext" != "xno"; then

